horovod/common/elastic.py

```diff
@@ -78,19 +79,26 @@ def check_host_updates(self):
         # Iterate through the update messages sent from the server. If the update timestamp
         # is greater than the last update timestamp, then trigger a HostsUpdatedException.
         last_updated_timestamp = prev_timestamp = self._last_updated_timestamp
+        last_updated_is_shrink = False
         while not self._host_messages.empty():
-            timestamp = self._host_messages.get()
+            timestamp,is_shrink = self._host_messages.get()
             if timestamp > last_updated_timestamp:
                 last_updated_timestamp = timestamp
+                last_updated_is_shrink = is_shrink
```

Comment:
There's something tricky here we need to watch out for: if there are multiple host updates sent by the driver between calls to this function, like:

```
host1 added
host2 removed
```

Then we will register `last_updated_is_shrink` as `True`.  But in this case we don't want to treat this as a shrink, because there was also a worker added.

Instead, we need to make sure that ALL of the host updates were shrinks (or, conversely, none of the host updates were adds).

Reply:
I fixed it by taking bitwise OR for all update history.